From: jiyun_ha <jiyun_ha@tmax.co.kr>
Date: Tue, 8 Jun 2021 17:09:43 +0900
Subject: [feat] tos-hotkey : close all hotkey widget when extension is
 disabled

 - remove signal handler for SessionMode change
 - make widget closed when extension is disabled

[IMS] N/A
---
 extensions/tos-hotkey/baseWidget.js | 2 --
 extensions/tos-hotkey/extension.js  | 6 ++++++
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/extensions/tos-hotkey/baseWidget.js b/extensions/tos-hotkey/baseWidget.js
index 5dddffe..79fb463 100644
--- a/extensions/tos-hotkey/baseWidget.js
+++ b/extensions/tos-hotkey/baseWidget.js
@@ -163,8 +163,6 @@ var BaseWidget = GObject.registerClass({ // eslint-disable-line no-unused-vars
 
         this._grabHelper = new GrabHelper.GrabHelper(this);
         Main.layoutManager.connect('system-modal-opened', () => this.close());
-
-        Main.sessionMode.connect('updated', () => this.close());
     }
 
     get state() {
diff --git a/extensions/tos-hotkey/extension.js b/extensions/tos-hotkey/extension.js
index a2df277..bb7489d 100644
--- a/extensions/tos-hotkey/extension.js
+++ b/extensions/tos-hotkey/extension.js
@@ -103,4 +103,10 @@ function disable() { // eslint-disable-line no-unused-vars
         Shell.ActionMode.NORMAL | Shell.ActionMode.OVERVIEW,
         Main.wm._startSwitcher.bind(Main.wm)
     );
+
+    if (systemWidget)
+        systemWidget.close();
+
+    if (switchMonitor)
+        switchMonitor.close();
 }
